rasterio/features.py

```diff
@@ -260,25 +264,24 @@ def format_cast_error(param, dtype):
             value = default_value
         geom = getattr(geom, '__geo_interface__', None) or geom
 
-        # geom must be a valid GeoJSON geometry type and non-empty
-        if not is_valid_geom(geom):
-            raise ValueError(
-                'Invalid geometry object at index {0}'.format(index)
-            )
-
-        if geom['type'] == 'GeometryCollection':
-            # GeometryCollections need to be handled as individual parts to
-            # avoid holes in output:
-            # https://github.com/mapbox/rasterio/issues/1253.
-            # Only 1-level deep since GeoJSON spec discourages nested
-            # GeometryCollections
-            for part in geom['geometries']:
-                valid_shapes.append((part, value))
+        if is_valid_geom(geom):
+            shape_values.append(value)
 
-        else:
-            valid_shapes.append((geom, value))
+            if geom['type'] == 'GeometryCollection':
+                # GeometryCollections need to be handled as individual parts to
+                # avoid holes in output:
+                # https://github.com/mapbox/rasterio/issues/1253.
+                # Only 1-level deep since GeoJSON spec discourages nested
+                # GeometryCollections
+                for part in geom['geometries']:
+                    valid_shapes.append((part, value))
 
-        shape_values.append(value)
+            else:
+                valid_shapes.append((geom, value))
+        
+        else:
+            # invalid or empty geometries are skipped and raise a warning instead
+            warnings.warn('SKIPPING: Invalid or empty geometry at index {}'.format(index))
```

Comment:
@brendan-ward first: it's great to see you here again! Second: how about "Invalid or empty shape at index {} will not be rasterized" as a message. In some situations, "SKIPPING" may be ambiguous.